src/EnergyPlus/PlantLoadProfile.cc

```diff
@@ -126,38 +123,79 @@ void PlantProfileData::simulate(EnergyPlusData &state,
+                                                                          this->InletTemp,
+                                                                          1.0,
+                                                                          state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                          RoutineName);
+            Real64 EnthSteamOutWet = FluidProperties::GetSatEnthalpyRefrig(state,
+                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
+                                                                           this->InletTemp,
+                                                                           0.0,
+                                                                           state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                           RoutineName);
+            Real64 LatentHeatSteam = EnthSteamInDry - EnthSteamOutWet;
+            Real64 SatTemp = FluidProperties::GetSatTemperatureRefrig(state,
+                                                                      state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
+                                                                      DataEnvironment::StdPressureSeaLevel,
+                                                                      state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                      RoutineName);
+            Real64 CpWater = FluidProperties::GetSpecificHeatGlycol(state,
+                                                                    state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidName,
+                                                                    SatTemp,
+                                                                    state.dataPlnt->PlantLoop(this->plantLoc.loopNum).FluidIndex,
+                                                                    RoutineName);
+
+            // Steam Mass Flow Rate Required
+            this->MassFlowRate = this->Power / (LatentHeatSteam + this->DegOfSubcooling * CpWater);
+
+            PlantUtilities::SetComponentFlowRate(state, this->MassFlowRate, this->InletNode, this->OutletNode, this->plantLoc);
+
+            // In practice Sensible & Superheated heat transfer is negligible compared to latent part.
+            // This is required for outlet water temperature, otherwise it will be saturation temperature.
+            // Steam Trap drains off all the Water formed.
+            // Here Degree of Subcooling is used to calculate hot water return temperature.
+
+            // Calculating Water outlet temperature
+            this->OutletTemp = SatTemp - this->LoopSubcoolReturn;
+        } else {
+            this->Power = 0.0;
+        }
```

Comment:
There is no OutletSteamQuality here? as there is in SteamCoils and OutsideEnergySources. Actually it's pretty much a useless variable since it's either 1 or 0.

Reply:
Thank you for catching this. I'll set OutletSteamQuality here.

Reply:
Or we might remove `OutletSteamQuality` here and from `DistrictHeating:Steam`. In the case of `Boiler:Steam`, it does not have component member variables about steam quality, and it directly updates the steam quality data of LoopNode like below.
`state.dataLoopNodes->Node(BoilerOutletNode).Quality = 1.0; `